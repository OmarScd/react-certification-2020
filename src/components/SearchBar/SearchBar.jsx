import React from 'react';
import styled from 'styled-components';

import { useSearch } from '../../providers/SearchContext';

import searchImg from '../../images/search-24px.svg';

const Form = styled.form`
  border-radius: 3px;
  display: flex;
  margin: 0;
  overflow: hidden;
`;

const Input = styled.input`
  border: none;
  outline: none;
  padding: 0 0.5rem;
`;

const SearchIcon = styled.img`
  background: white;
  margin: 0;
`;

const SearchBar = ({ handleOnSubmit }) => {
  const { query, setQuery } = useSearch();

  return (
    <Form onSubmit={handleOnSubmit}>
      <SearchIcon src={searchImg} alt="magnifying lens" />
      <Input
        type="text"
        placeholder="Search..."
        value={query}
        onChange={(e) => setQuery(e.target.value)}
      />
    </Form>
  );
};

export default SearchBar;
