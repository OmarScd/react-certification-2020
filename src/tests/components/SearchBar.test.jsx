import React from 'react';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import SearchBar from '../../components/SearchBar';
import SearchContextProvider from '../../providers/SearchContext';

describe('SearchBar', () => {
  afterAll(cleanup);

  it('renders correctly', () => {
    const { container } = render(
      <SearchContextProvider>
        <SearchBar />
      </SearchContextProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('updates input text on change', () => {
    const query = 'hola';
    render(
      <SearchContextProvider>
        <SearchBar />
      </SearchContextProvider>
    );
    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: query } });
    expect(screen.getByDisplayValue(query)).toBeTruthy();
  });

  it('searches on submit', () => {
    const handleOnSubmit = jest.fn();
    render(
      <SearchContextProvider>
        <SearchBar handleOnSubmit={handleOnSubmit} />
      </SearchContextProvider>
    );
    const input = screen.getByRole('textbox');
    fireEvent.submit(input);
    expect(handleOnSubmit).toHaveBeenCalled();
  });
});
