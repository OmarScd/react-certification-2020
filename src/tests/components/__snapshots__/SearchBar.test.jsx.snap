// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<form
  class="sc-bdnylx hYwFxl"
>
  <img
    alt="magnifying lens"
    class="sc-dlnjPT dGgzTM"
    src="search-24px.svg"
  />
  <input
    class="sc-gtssRu fkigkU"
    placeholder="Search..."
    type="text"
    value="Subdivisions"
  />
</form>
`;
